src/main/java/seedu/address/model/tag/TagName.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.model.tag;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Tag's name in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ */
+public class TagName {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
```

Comment:
I think maybe can change this to "Tag names should..." just to make it clear what you are referring to

Reply:
Thanks for mentioning, i'll make the changes